src/fontra/views/fontinfo/panel-sources.js

```diff
@@ -1,14 +1,604 @@
+      0
+    );
+
+    validateInput();
+
+    if (!(await dialog.run())) {
+      // User cancelled
+      return;
+    }
+
+    let newLocation = makeSparseLocation(locationController.model, locationAxes);
+    for (const axis of locationAxes) {
+      if (!(axis.name in newLocation)) {
+        newLocation[axis.name] = axis.defaultValue;
+      }
+    }
+
+    const interpolatedSource = getInterpolatedSourceData(
+      this.fontController,
+      newLocation
+    );
+    // TODO: round the interpolated vertical metrics to 2 decimal places
+    const newSource = {
+      name: nameController.model.sourceName.trim(),
+      italicAngle: nameController.model.sourceItalicAngle,
+      location: newLocation,
+    };
+    return {
+      verticalMetrics: getDefaultVerticalMetrics(this.fontController.unitsPerEm),
+      ...interpolatedSource,
+      ...newSource,
+    };
+  }
+
+  getSourceName(sources) {
+    const sourceNames = Object.keys(sources).map(function (sourceIdentifier) {
+      return sources[sourceIdentifier].name;
+    });
+    let sourceName = "Untitled source";
+    let i = 1;
+    while (sourceNames.includes(sourceName)) {
+      sourceName = `Untitled source ${i}`;
+      i++;
+    }
+    return sourceName;
+  }
+
+  _sourcePropertiesContentElement(locationAxes, nameController, locationController) {
+    const locationElement = html.createDomElement("designspace-location", {
+      style: `grid-column: 1 / -1;
+        min-height: 0;
+        overflow: auto;
+        height: 100%;
+      `,
+    });
+    locationElement.axes = locationAxes;
+    locationElement.controller = locationController;
+
+    const containerContent = [
+      ...labeledTextInput("Source name:", nameController, "sourceName", {}),
+      ...labeledTextInput("Italic Angle:", nameController, "sourceItalicAngle", {}),
+      html.br(),
+      locationElement,
+    ];
+
+    const warningElement = html.div({
+      id: "warning-text",
+      style: `grid-column: 1 / -1; min-height: 1.5em;`,
+    });
+    containerContent.push(warningElement);
+
+    const contentElement = html.div(
+      {
+        style: `overflow: hidden;
+          white-space: nowrap;
+          display: grid;
+          gap: 0.5em;
+          grid-template-columns: max-content auto;
+          align-items: center;
+          height: 100%;
+          min-height: 0;
+        `,
+      },
+      containerContent
+    );
+
+    return { contentElement, warningElement };
+  }
+}
+
+addStyleSheet(`
+:root {
+  --fontra-ui-font-info-sources-panel-max-list-height: 12em;
+}
+
+.fontra-ui-font-info-sources-panel-source-box {
+  background-color: var(--ui-element-background-color);
+  border-radius: 0.5em;
+  padding: 1em;
+  cursor: pointer;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: max-content max-content max-content max-content auto;
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+  max-height: 70px;
+}
+
+.fontra-ui-font-info-column {
```

Comment:
Please use the proper _full_ prefix:
`fontra-ui-font-info-sources-panel- ...`

Same in the rest of the classes below.

These classes are global, and therefore _must_ be unique, so we add a big fat prefix.

Reply:
done.